### ISSUE TITLE
all() does not respect lane mask

### ISSUE THREAD

**Issue by caughtbyatoe** [ispc/ispc#1317]
It looks like all() examines all lanes regardless if those lanes are active or not.  For example,

```
if (programIndex < 4) {
        if (all(programIndex < 4)) {
            print("all lanes are less than 4\n");
        } else {
            print("all lanes are not less than 4\n");
        }
}
```
will output:

all lanes are not less than 4

This behavior is usually inconvenient.  When using all(), I'm generally thinking all of my active lanes - the inactive lanes often contain irrelevant values.

**Comment by DeepakRajendrakumaran** [comment]
This has been fixed with the following commit: https://github.com/ispc/ispc/commit/d6fb646e589ae12b229f86f4eaa86d12ffb48d10

**Comment by dbabokin** [comment]
Fixed in PR #1333
